eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1722,7 +1724,7 @@ protected function copyTranslationsFromPublishedVersion(APIVersionInfo $versionI
 
             if ($newValue !== null
                 && $field->value !== null
-                && $fieldType->toHash($newValue) === $fieldType->toHash($field->value)) {
+                && $this->isHashEqual($fieldType, $newValue, $field->value)) {
```

Comment:
It's not a responsibility of Content Service to know details of equality conditions of a field type. It should be exposed s you're able to do `$newValue->equals($field->value)` (or following my next comment: `$fieldType->valuesEqual($newValue, $field->value)`).